pkg/ipam/types/types.go

```diff
@@ -56,6 +57,18 @@ type AllocationMap map[string]AllocationIP
 // +kubebuilder:validation:Format=cidr
 type IPAMPodCIDR string
 
+func (c *IPAMPodCIDR) ToPrefix() (*netip.Prefix, error) {
+	if c == nil {
+		return nil, fmt.Errorf("nil ipam cidr")
+	}
+
+	if prefix, err := netip.ParsePrefix(string(*c)); err == nil {
+		return &prefix, nil
+	}
+
+	return nil, fmt.Errorf("failed to parse ipam cidr %v", c)
```

Comment:
The error is not reported in the return here.

Reply:
@christarazi thanks for the review! Commit d6c79cf resolves this issue.